### PR TITLE
Fix WhitespaceTokenizer throwing ValueError

### PR DESCRIPTION
**Pull request by tabergma**
**Proposed changes**:
- Add `not_supported_language_list` to component to be able to exclude language that the component can handle
- `WhitespaceTokenizer` only removes emoji if the complete token is matching the regex

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by tabergma** [comment]
Differences of the old method (before we switched to `regex`) and the code in this PR:
[whitespace_tokenizer.txt](https://github.com/RasaHQ/rasa/files/4883981/whitespace_tokenizer.txt)

File: rasa/nlu/components.py [hunk 1]
```diff
@@ -648,10 +654,15 @@ def can_handle_language(cls, language: Hashable) -> bool:
...
+        if language is None or (
+            cls.language_list is None and cls.not_supported_language_list is None
+        ):
             return True
 
-        return language in cls.language_list
+        language_list = cls.language_list or []
+        not_supported_language_list = cls.not_supported_language_list or []
+
+        return language in language_list or language not in not_supported_language_list
```
Comment (dakshvar22) [inline rasa/nlu/components.py:665]: Can we refactor `language_list` variable to `supported_language_list`? Took me a while to decipher what all `language` variables were there 😄

File: changelog/6150.feature.rst [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Add ``not_supported_language_list`` to component to be able to define languages that a component can NOT handle.
+
+``WhitespaceTokenizer`` is not able to process languages which are not separated by whitespace. ``WhitespaceTokenizer``
+will throw an error if it is used with Chinese, Japanese, Korean, Thai, and Vietnamese.
```
Comment (tabergma) [inline changelog/6150.feature.rst:4]: @dakshvar22 Just to be sure: Any other language to add or should I remove any of those languages?

Comment (dakshvar22) [reply changelog/6150.feature.rst:4]: I also wasn't sure about it. This link seems to answer our question - https://linguistics.stackexchange.com/questions/6131/is-there-a-long-list-of-languages-whose-writing-systems-dont-use-spaces

Comment (tabergma) [reply changelog/6150.feature.rst:4]: thanks, so will remove Korean and Vietnamese from the list

**Pull request merged**